From: Markus Lehtonen <markus.lehtonen@linux.intel.com>
Date: Wed, 24 Apr 2013 14:48:54 +0300
Subject: Fedora HACK: fix debchange

For older Fedora versions.

Gbp-Rpm: If 0%{?fedora_version} && 0%{?fedora_version} < 19

Change-Id: I99eb609f826d0dc459fd719fd5ed87535919eea1
Signed-off-by: Markus Lehtonen <markus.lehtonen@linux.intel.com>
---
 scripts/debchange.pl | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/scripts/debchange.pl b/scripts/debchange.pl
index a5c3936..ea52288 100755
--- a/scripts/debchange.pl
+++ b/scripts/debchange.pl
@@ -56,6 +56,39 @@ my $CHGLINE;  # used by the format O section at the end
 
 my $lpdc_broken;
 
+#
+# Backported from newer Dpkg::Compression
+#
+my $COMP = {
+    "gzip" => {
+	"file_ext" => "gz",
+	"comp_prog" => [ "gzip", "--no-name", "--rsyncable" ],
+	"decomp_prog" => [ "gunzip" ],
+	"default_level" => 9,
+    },
+    "bzip2" => {
+	"file_ext" => "bz2",
+	"comp_prog" => [ "bzip2" ],
+	"decomp_prog" => [ "bunzip2" ],
+	"default_level" => 9,
+    },
+    "lzma" => {
+	"file_ext" => "lzma",
+	"comp_prog" => [ 'xz', '--format=lzma' ],
+	"decomp_prog" => [ 'unxz', '--format=lzma' ],
+	"default_level" => 6,
+    },
+    "xz" => {
+	"file_ext" => "xz",
+	"comp_prog" => [ "xz" ],
+	"decomp_prog" => [ "unxz" ],
+	"default_level" => 6,
+    },
+};
+
+my $regex = join "|", map { $_->{"file_ext"} } values %$COMP;
+our $compression_re_file_ext = qr/(?:$regex)/;
+
 sub have_lpdc {
     return ($lpdc_broken ? 0 : 1) if defined $lpdc_broken;
     eval {
